rte/mo_util_array.F90

```diff
@@ -34,73 +34,46 @@ module mo_util_array
   !-------------------------------------------------------------------------------------------------
   ! Values less than a floor
   !-------------------------------------------------------------------------------------------------
-  logical function any_vals_less_than_1D(array, minVal)
+  logical function any_vals_less_than_1D(array, check_value)
     real(wp), dimension(:), intent(in) :: array
-    real(wp),               intent(in) :: minVal
+    real(wp),               intent(in) :: check_value
 
-#ifdef _OPENACC
-    ! Compact version using intrinsics below
-    ! but an explicit loop is the only current solution on GPUs
     real(wp) :: minValue
-    integer  :: i
 
-    minValue = minVal
-    !$acc parallel loop copyin(array) reduction(min:minValue)
-    do i = 1, size(array,1)
-      minValue = min(array(i), minValue)
-    end do
-    any_vals_less_than_1D = (minValue < minVal)
-#else
-    any_vals_less_than_1D = any(array < minVal)
-#endif
-end function any_vals_less_than_1D
-  !-------------------------------------------------------------------------------------------------
-  logical function any_vals_less_than_2D(array, minVal)
+    !$acc kernels copyin(array)
+    minValue = minval(array)
```

Comment:
Oooh, I hadn't realized you could support intrinsics like minval on the GPU. That's much cleaner.

Reply:
Yeah, I agree!